### PR TITLE
Improve developer documentation

### PR DESCRIPTION
**Pull request by WanjaTschuorSICKAG**

### CODE REVIEW

File: src/lib/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+The `expose` function is the only function which is exported directly to be used in the micro-frontends.
+
+The other functions are for setting up a collage context and performing the handshake between Fragment and Arrangement.
+
+## logging
+Module for internal logging
+
+## messages
+When establishing the connection between a fragment and its arrangement, penpal is not available for communication jet. So we are using the postMessage Api directly for performing a handshake.
```
Comment (gintephSICKAG) [inline src/lib/README.md:23]: "yet" instead of "jet"

File: src/lib/api/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# API
+
+Types and Syntactic Suggar for collage
```
Comment (gintephSICKAG) [inline src/lib/api/README.md:3]: "Sugar"

File: src/lib/collage-plugin.ts [hunk 3]
```diff
@@ -29,8 +34,15 @@ export function mergeContexts(context: unknown, append: unknown) {
   });
 }
 
-// TODO: Find out if we can unify the buildContext etc functions
-
+/**
+ * Builds the context type returnd by a call of expose in a micro-frontend by merging the existing context
```
Comment (gintephSICKAG) [inline src/lib/collage-plugin.ts:38]: "returned" instead of "returnd"

File: src/lib/collage-plugin.ts [hunk 4]
```diff
@@ -29,8 +34,15 @@ export function mergeContexts(context: unknown, append: unknown) {
...
+ * type with a plugin specific context type. This happens one after the other for each plugin.
+ *
+ * @param previous context before the plugin is called
+ * @param pluginFunction function returning the plugin specific context type and functions
+ * @returns the exposed function to be used in the micro-frontend.
+ * The expoesed function then returns the merged context type.
```
Comment (gintephSICKAG) [inline src/lib/collage-plugin.ts:44]: "exposed"

File: src/lib/messages.ts [hunk 5]
```diff
@@ -42,6 +45,9 @@ export function listenFor(
   window.addEventListener('message', listenerCallback);
 }
 
+/**
+ * Send a postMessage event to the given recepient
```
Comment (gintephSICKAG) [inline src/lib/messages.ts:49]: rec_i_pient

**Review by gintephSICKAG — COMMENTED** [review]
Looks good :)

**Comment by WanjaTschuorSICKAG** [comment]
Thanks for reviewing @gintephSICKAG

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**